pkg/querier/store_gateway_client.go

```diff
@@ -0,0 +1,52 @@
+package querier
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/cortexproject/cortex/pkg/ring/client"
+	"github.com/cortexproject/cortex/pkg/storegateway/storegatewaypb"
+	"github.com/cortexproject/cortex/pkg/util/grpcclient"
+)
+
+func NewStoreGatewayClientFactory(cfg grpcclient.Config, reg prometheus.Registerer) client.PoolFactory {
```

Comment:
This doesn't look to be called anywhere? I assume it will be added in a consecutive PR.

Reply:
Exactly. I'm still in the process of submitting partial PRs in preparation of the final PR where I will submit the store-gateway and blocks sharding logic, otherwise the final one will be too huge to review.